crates/kitsune_p2p/proxy/src/bin/proxy-cli.rs

```diff
@@ -0,0 +1,80 @@
+use futures::stream::StreamExt;
+use kitsune_p2p_proxy::*;
+use kitsune_p2p_transport_quic::*;
+use kitsune_p2p_types::{dependencies::ghost_actor, transport::*};
+use structopt::StructOpt;
+
+/// Option Parsing
+#[derive(structopt::StructOpt, Debug)]
+#[structopt(name = "proxy-cli")]
+pub struct Opt {
+    /// kitsune-proxy Url to connect to.
+    pub proxy_url: String,
+
+    /// If you would like to keep pinging, set an interval here.
+    #[structopt(short = "t", long)]
+    pub time_interval_ms: Option<u64>,
+}
+
+#[tokio::main]
+async fn main() {
+    let _ = ghost_actor::dependencies::tracing::subscriber::set_global_default(
+        tracing_subscriber::FmtSubscriber::builder()
+            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .finish(),
+    );
+
+    if let Err(e) = inner().await {
+        eprintln!("{:?}", e);
+    }
+}
+
+async fn inner() -> TransportResult<()> {
+    let opt = Opt::from_args();
+
+    let (listener, events) = spawn_transport_listener_quic(ConfigListenerQuic::default()).await?;
+
+    let proxy_config = ProxyConfig::local_proxy_server(
+        TlsConfig::new_ephemeral().await?,
+        AcceptProxyCallback::reject_all(),
+    );
+
+    let (listener, mut events) =
+        spawn_kitsune_proxy_listener(proxy_config, listener, events).await?;
+
+    tokio::task::spawn(async move {
+        while let Some((url, mut write, read)) = events.next().await {
+            eprintln!("# ERR incoming msg from {}", url);
+            drop(read);
+            let _ = write.write_and_close(Vec::with_capacity(0)).await;
+        }
+    });
+
+    println!(
+        "# SELF URL (you can ignore this : ) {}",
```

Comment:
Why can we ignore this? Is this a magic string that gets parsed elsewhere? Would be good to put in a comment about the implications of this line.

Reply:
haha, totally, yeah I can add some comments - or just delete this printout... the client itself needs to bind a quic port in order to talk to the server... but nobody will ever use it, it's only going to make the one outgoing connection.